File: apps/spine/lib/drift/model/sales_item.dart
import 'package:drift/drift.dart';
import 'package:spine/drift/model/base.dart';
import 'package:spine/drift/model/product.model.dart';
import 'package:spine/drift/model/sales.model.dart';
import 'package:spine/drift/model/batch.model.dart';

class SalesItem extends Table with TableMixin {
  late final name = text()();
  late final quantity = integer().withDefault(const Constant(1))();
  late final type = text()();// product, charge
  late final unitPrice = integer()();
  late final costPrice = integer()();
  late final total = integer()();
  late final description = text().nullable()();

  late final saleId = text().references(Sales, #id)();
  late final productId = text().nullable().references(Product, #id)();
  late final batchId = text().nullable().references(SpineBatch, #id)();
}

File: packages/shared/src/types/sales-item.ts
import { BaseModel, SyncableModel } from "./base-model-type";

export interface ISalesItem extends BaseModel, SyncableModel {
    name: string,
    quantity: number;
    type: string,
    unit_price: number,
    cost_price: number,
    total: number,
    description: string,

    sale_id: string,
    product_id?: string,
    batch_id?: string,

}